core/src/main/java/org/opensearch/sql/analysis/HighlightAnalyzer.java

```diff
@@ -34,8 +34,10 @@ public LogicalPlan visitAlias(Alias node, AnalysisContext context) {
       return null;
     }
 
+    // Must set name here else operator cannot resolve delegated reference
+    ((HighlightFunction) node.getDelegated()).setName(node.getName());
```

Comment:
Can you use `unresolved` and call `setName` after it is initialized?

Reply:
`unresolved` will be initialized with a null name in this case.